File: Time_Conversion.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'timeConversion' function below.
#
# The function is expected to return a STRING.
# The function accepts STRING s as parameter.
#

def timeConversion(s):
    hh,mm,ss = s[0:2],s[3:5],s[6:8]
    typ = s[-2:]
    if hh == "12" and typ == "PM":
        hh = "12"
    elif hh == "12" and typ == "AM":
        hh = "00"
    elif typ == "PM":
        hh = int(hh) + 12
    return (str(hh) + ":" + mm + ":" + ss)

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    result = timeConversion(s)

    fptr.write(result + '\n')

    fptr.close()

File: Time_Conversion.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'timeConversion' function below.
     *
     * The function is expected to return a STRING.
     * The function accepts STRING s as parameter.
     */

    public static String timeConversion(String s) {
        String hours = s.substring(0,2);
        String minutes = s.substring(3,5);
        String seconds = s.substring(6,8);
        String meridiem = s.substring(8);
        
        if (meridiem.equals("AM")) {
            if (hours.equals("12")) {
                hours = "00";
            }
        } else { 
            if (!hours.equals("12")) {
                hours = String.valueOf(
                    Integer.parseInt(hours) + 12
                );
            }
        }
        
        return hours + ":" + minutes + ":" + seconds;
        
    }

}

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        String s = bufferedReader.readLine();

        String result = Result.timeConversion(s);

        bufferedWriter.write(result);
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }
}
